### PR TITLE
Task/refactor color service

### PR DESCRIPTION
**Pull request by esotrope**
started added alt scores to the student view and got side tracked standardizing the way we get colors for assessment types and grade codes

### CODE REVIEW

File: webapp/src/main/webapp/src/app/dashboard/group-dashboard/group-assessment-card.component.ts [hunk 1]
```diff
@@ -18,68 +23,50 @@ export interface AssessmentCardEvent {
 
 @Component({
   selector: 'group-assessment-card',
-  templateUrl: './group-assessment-card.component.html'
+  templateUrl: './group-assessment-card.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/dashboard/group-dashboard/group-assessment-card.component.ts]: optimized group and student card views in the process

File: webapp/src/main/webapp/src/app/student/results/student-results.component.ts [hunk 2]
```diff
@@ -164,13 +162,6 @@ export class StudentResultsComponent implements OnInit {
     });
   }
 
-  getAssessmentTypeColor(assessmentType: string): string {
-    const index = ['ica', 'iab', 'sum'].indexOf(assessmentType);
-    const totalAssessmentTypes = 3;
-    const colorIndex = index >= 0 ? index + 1 : totalAssessmentTypes;
-    return this.colorService.getColor(colorIndex);
-  }
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/student/results/student-results.component.ts]: extracted to `colors.ts`

File: webapp/src/main/webapp/src/app/shared/colors.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  'green-dark',
+  'blue-dark'
+];
+
+export function color(index: number): string {
+  return colors[index % colors.length];
+}
+
+export function gradeColor(gradeCode: string): string {
+  const index = gradeCodes.indexOf(gradeCode);
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/shared/colors.ts]: Does this make it so the `grade.xx.color` values from common en.json are no longer needed?

File: webapp/src/main/webapp/src/app/shared/colors.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  return color(index >= 0 ? index : 0);
+}
+
+export function assessmentTypeColor(assessmentType: AssessmentType): string {
+  const index = assessmentTypes.indexOf(assessmentType);
```
Comment (malaffoon) [inline webapp/src/main/webapp/src/app/shared/colors.ts]: I like the indirection approach.

File: webapp/src/main/webapp/src/app/shared/colors.ts (hunk 3 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/app/shared/colors.ts]: hm.. wasn't aware of their presence. My guess is those are being used for server side PDFs and this was used on the UI. I'm thinking now we should only be using en.json...

Comment (esotrope) [reply webapp/src/main/webapp/src/app/shared/colors.ts]: The upside to doing it in the code is we can apply fallback colors if the grade code or whatever is absent for some reason in the translation file this however should never be a real case though so i'm leaning toward single source of truth for this

**Pull request merged**